### ISSUE TITLE
F.cross_entropy returns infs sometimes due to it summing the losses.

### ISSUE THREAD

**Issue by aced125** [microsoft/DeepSpeed#962]
Hello,

F.cross_entropy returns infs sometimes due to it summing the losses.

Findings:

The following SOMETIMES returns inf loss (the default options in F.cross_entropy)
```python
F.cross_entropy(preds, labels, ignore_index=-100, reduction="mean")
```

The following will NOT return inf
```python
loss = F.cross_entropy(preds, labels, ignore_index=-100, reduction="none")
loss = loss.mean()
```

The following SOMETIMES returns inf
```python
loss = F.cross_entropy(preds, labels, ignore_index=-100, reduction="none")
loss = loss.sum()
```

The solution I am currently using to do normal cross entropy in DeepSpeed:

```python
loss = F.cross_entropy(logits, labels, reduction="none")
numel = labels.numel()
numel_no_mask = labels.ne(-100).sum()
norm = numel_no_mask / numel

loss = loss.mean() / norm
```

**Comment by cli99** [comment]
Can you clarify how the problem relates to DeepSpeed? Or can you share the full reproduce code of the problem? Thanks.

**Comment by Mindful** [comment]
I'm not sure I understand what's going on here or exactly how this relates to deepspeed, but I think I may just have encountered the same problem. I found that increasing my batch size while using deepspeed resulted in `inf` loss - I changed only batch size, nothing else, so I really don't understand how this could happen - but replacing NLL loss with reduction="mean" with reduction="none" and then `.mean()` appears to have solved the issue. 

I did not see this issue in pure Pytorch training, and I only saw it with deepspeed when I pushed to batch sizes larger than what I could handle without using deepspeed.
